waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -33,11 +38,12 @@ proc init*(T: type WakuMessageStore, db: SqliteDatabase): MessageStoreResult[T]
   let prepare = db.prepareStmt("""
     CREATE TABLE IF NOT EXISTS """ & TABLE_TITLE & """ (
```

Comment:
Could this `CREATE TABLE` become part of the migration scripts? I guess it depends on how important it is being able to set the table name, right?

Reply:
Indeed it can be part of the migration scripts and I cannot think of any issue why it should not be. @oskarth @jm-clius wdyt? if there is no concern, then I can separate it as a migration script.

Reply:
Re the table name, you are right, however, it is currently hardcoded in the code and is not a user input. As such, effectively there is no difference in having it as part of the migration scripts or in the code.

Reply:
It could be done, but I would prefer it if it wasn't part of the migration. Since the SQLite tables should only be created if the node is configured to do so (i.e. if `persistMessages` or `persistPeers` are enabled), the migration would have to be aware of these config items to decide whether or not the `CREATE TABLE` should be run. In my view the order in which things should happen:
1. Node starts
2. Node checks if it has an existing SQLite DB (`dbPath`)
3. If true, node migrates DB to desired version (with each script only running for tables that already exists).
4. Node creates new tables based on configuration.

This is more or less what is already being done, with only (3) requiring modification to the scripts.

Reply:
> Since the SQLite tables should only be created if the node is configured to do so (i.e. if persistMessages or persistPeers are enabled), the migration would have to be aware of these config items to decide whether or not the CREATE TABLE should be run.

In the current implementation, the migration is done only if the waku node is a full store node and is instructed to persist messages (if the configuration indicates so). It means the call to the migrate proc is already tied to the `persistMessages` (or `persistPeers`) flags. No migration will take place (hence no table will be created) if these flags are not true.

Reply:
>If DB exists, node migrates DB to desired version (with each script only running for tables that already exists)

I understand your point.  However, this approach of "running scripts only when the corresponding table exists", is subject to error e.g., when the table is supposed to exist but is externally deleted from the DB. 

Right now we can perfectly distinguish between peer store migration flow and message store migration part. For this sake, the `migrate` procedure takes a path to the directory containing the migration scripts and starts running them.  The migration scripts of the message store and the peer store are currently maintained in two separate folders exactly for this purpose. As such, the migration procedure would be like this
- Check `persistMessages`, if set then call `migrate(<path to the migration scripts of the message store>)`
- Check `persistPeers `, if set then call `migrate(<path to the migration scripts of the peer store>)`

Please let me know whether this addresses your concern.

Reply:
Yeah, thanks, this works for me.